Avoid calls to deprecated __glp_lib_* functions which are deprecated
and have been removed for the GLPK public API since version 4.32.

Taken from the Mercurial usptream repostory:
http://hg.savannah.gnu.org/hgweb/octave/raw-diff/aaccb1362347/src/DLD-FUNCTIONS/__glpk__.cc

 -- Rafael Laboissiere <rafael@debian.org>  Wed, 04 Feb 2009 11:56:17 +0100

--- a/src/DLD-FUNCTIONS/__glpk__.cc	Wed Aug 27 09:59:55 2008 -0400
+++ b/src/DLD-FUNCTIONS/__glpk__.cc	Mon Oct 27 14:25:30 2008 -0400
@@ -45,6 +45,7 @@
 #include <glpk.h>
 #endif
 
+#if 0
 #ifdef GLPK_PRE_4_14
 
 #ifndef _GLPLIB_H
@@ -62,6 +63,7 @@
 void _glp_lib_print_hook (int (*func)(void *info, char *buf), void *info);
 void _glp_lib_fault_hook (int (*func)(void *info, char *buf), void *info);
 
+#endif
 #endif
 }
 
@@ -137,6 +139,7 @@
 
 static jmp_buf mark;  //-- Address for long jump to jump to
 
+#if 0
 int
 glpk_fault_hook (void * /* info */, char *msg)
 {
@@ -150,6 +153,7 @@
   message (0, "%s", msg);
   return 1;
 }
+#endif
 
 int
 glpk (int sense, int n, int m, double *c, int nz, int *rn, int *cn,
@@ -164,6 +168,7 @@
 
   clock_t t_start = clock();
 
+#if 0
 #ifdef GLPK_PRE_4_14
   lib_set_fault_hook (0, glpk_fault_hook);
 #else
@@ -175,6 +180,7 @@
     lib_set_print_hook (0, glpk_print_hook);
 #else
     _glp_lib_print_hook (glpk_print_hook, 0);
+#endif
 #endif
 
   LPX *lp = lpx_create_prob ();
@@ -314,11 +320,14 @@
       break;
 
     default:
+      break;
+#if 0
 #ifdef GLPK_PRE_4_14
       insist (method != method);
 #else
       static char tmp[] = "method != method";
       glpk_fault_hook (0, tmp);
+#endif
 #endif
     }
 




